packages/typespec-client-generator-core/test/decorators.test.ts

```diff
@@ -3785,4 +3785,282 @@ describe("typespec-client-generator-core: decorators", () => {
+      strictEqual(methods.length, 1);
+      const download = methods[0];
+      strictEqual(download.name, "download");
+      strictEqual(download.kind, "basic");
+      strictEqual(download.parameters.length, 1);
+      strictEqual(download.parameters[0].name, "blobName");
+      strictEqual(download.parameters[0].onClient, true);
+
+      const downloadOp = download.operation;
+      strictEqual(downloadOp.parameters.length, 1);
+      const blobNameOpParam = downloadOp.parameters[0];
+      strictEqual(blobNameOpParam.name, "blobName");
+      strictEqual(blobNameOpParam.correspondingMethodParams.length, 1);
+      strictEqual(blobNameOpParam.correspondingMethodParams[0], blobName);
+    });
+    it("subclient", async () => {
+      await runner.compileWithCustomization(
+        `
+        @service
+        namespace StorageClient {
+
+          @route("/main")
+          op download(@path blobName: string): void;
+
+          interface BlobClient {
+            @route("/blob")
+            op download(@path blobName: string): void;
+          }
+        }
+        `,
+        `
+        model BlobClientInitialization {
+          blobName: string
+        };
+
+        @@clientInitialization(StorageClient.BlobClient, BlobClientInitialization);
+        `
+      );
+      const sdkPackage = runner.context.sdkPackage;
+      const clients = sdkPackage.clients;
+      strictEqual(clients.length, 1);
+      const client = clients[0];
+      strictEqual(client.name, "StorageClient");
+      strictEqual(client.initialization.access, "public");
+      strictEqual(client.initialization.properties.length, 1);
+      strictEqual(client.initialization.properties[0].kind, "endpoint");
+
+      const methods = client.methods;
+      strictEqual(methods.length, 2);
+
+      // the main client's function should not have `blobName` as a client method parameter
+      const mainClientDownload = methods.find((x) => x.kind === "basic" && x.name === "download");
+      ok(mainClientDownload);
+      strictEqual(mainClientDownload.parameters.length, 1);
+      strictEqual(mainClientDownload.parameters[0].name, "blobName");
+      strictEqual(mainClientDownload.parameters[0].onClient, false);
+
+      const getBlobClient = methods.find((x) => x.kind === "clientaccessor");
+      ok(getBlobClient);
+      strictEqual(getBlobClient.kind, "clientaccessor");
+      strictEqual(getBlobClient.name, "getBlobClient");
+      strictEqual(getBlobClient.parameters.length, 1);
+      const blobNameParam = getBlobClient.parameters.find((x) => x.name === "blobName");
+      ok(blobNameParam);
+      strictEqual(blobNameParam.onClient, true);
+      strictEqual(blobNameParam.optional, false);
+      strictEqual(blobNameParam.kind, "method");
+
+      const blobClient = getBlobClient.response;
+
+      strictEqual(blobClient.kind, "client");
+      strictEqual(blobClient.name, "BlobClient");
+      strictEqual(blobClient.initialization.access, "public");
```

Comment:
my understanding was that `interface`s mapped to `operation groups` when no decorators were set, but it seems that the initialization is still public. That's totally fine, I jsut want to clarify my understanding @tadelesh. That's why I updated my code so that we only throw an error if there are new client params introduced for ogs explicitly titled with `@operationGroup`

Reply:
i think whether explicitly set or not, the behavior should be same. why op from interface infer could be public and `@operationGroup` be internal?

Reply:
i'm not sure, that's the current behavior though, so I think there's something wrong with current logic. Let me take a look today

Reply:
I kind of agree with Chenjie, that they be same.

A different scenario would be namespace nested under namespace (which is the subclient? -- do we have similar way to specify this with explicit `@client`)
```
namespace Client {
  namespace SubClient {
  }
}
```